### PR TITLE
Add semantic script, trainer

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR adds `run_semantic_segmentation.py`, which leverages the Trainer API for fine-tuning any model supported by `AutoModelForSemanticSegmentation`.

To do:

- [x] fix WandB logging, which doesn't seem to like the fact that metrics are turned into lists. Might need some help here from @sgugger.
- [x] also wondering why the [model card](https://huggingface.co/nielsr/segformer-finetuned-sidewalk-trainer) doesn't include the dataset name (even though it has a default value in the script):

Screenshot 2022-04-19 at 16 19 22

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this! It's not just Wandb, all loggers will complain about lists in metric values, so you should find a way to given them keys (at worst convert the list into a dict with `{i: v for i, vin enumerate(list_to_convert)}`).

Not sure where the dataset name is replaced by None, but normally it should be passed along to the model card.
